drivers/mil_passive_sonar/multilateration/multilateration.py

```diff
@@ -106,94 +217,96 @@ def get_B(delta):
             source = [0, 0, 0]
         return source
 
-    def estimate_pos_LS(self, timestamps):
+    def estimate_pos_LS(self, dtoa, cost_func):
         '''
-        Returns a ros message with the location and time of emission of a pinger pulse.
+        Uses the a minimization routine to solve for the position of a source base on dtoa measurements
         '''
-        self.timestamps = timestamps
+        self.dtoa = dtoa
         init_guess = np.random.normal(0,100,3)
         opt = {'disp': 0}
         opt_method = 'Powell'
-        result = optimize.minimize(self.cost_LS, init_guess, method=opt_method, options=opt, tol=1e-15)
+        result = optimize.minimize(cost_func, init_guess, method=opt_method, options=opt, tol=1e-15)
         if(result.success):
             source = [result.x[0], result.x[1], result.x[2]]
         else:
             source = [0, 0, 0]
         return source
 
     def cost_LS(self, potential_pulse):
-        '''
-        Compares the difference in observed and theoretical difference in time of arrival
-        between tevery unique pair of hydrophones.
-
-        Note: the result will be along the direction of the heading but not at the right distance.
-        '''
-        cost = 0
-        t = self.timestamps
-        c = self.c
-        x = np.array(potential_pulse)
-        for pair in self.pairs:
-            h0 = self.hydrophone_locations[pair[0]]
-            h1 = self.hydrophone_locations[pair[1]]
-            residual = la.norm(x-h0) - la.norm(x-h1) - c*(t[pair[0]] - t[pair[1]])
-            cost += residual**2
-        return cost
-
-    def cost_LS2(self, potential_pulse):
         """
         Slightly less accurate than the one above in terms of heading but much faster.
         """
         cost = 0
-        t = self.timestamps
-        x0 = self.hydrophone_locations[0][0]
-        y0 = self.hydrophone_locations[0][1]
-        z0 = self.hydrophone_locations[0][2]
+        t = self.dtoa
         x = potential_pulse[0]
         y = potential_pulse[1]
         z = potential_pulse[2]
-        d0 = np.sqrt((x0 - x)**2 + (y0 - x)**2 + (z0 - x)**2)
-        for i in xrange(1, len(self.hydrophone_locations)):
-            xi = self.hydrophone_locations[i][0]
-            yi = self.hydrophone_locations[i][1]
-            zi = self.hydrophone_locations[i][2]
+        d0 = np.sqrt((x)**2 + (x)**2 + (x)**2)
+        for i in range(self.n - 1):
+            xi = self.receiver_locations[i, 0]
+            yi = self.receiver_locations[i, 1]
+            zi = self.receiver_locations[i, 2]
             di = np.sqrt((xi - x)**2 + (yi - x)**2 + (zi - x)**2)
```

Comment:
Same with the `x`, `y`, `z`? If so this is the same as `np.linalg.norm` which would be more faster